payroll-processor-client/src/app/payroll/payroll-charts/payroll-charts.component.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Component,
+  OnInit,
+} from '@angular/core';
+
+import { DataService } from 'src/app/data/data-service';
+import { NameValue } from 'src/app/data/state/data-model';
+
+@Component({
+  selector: 'app-payroll-charts',
+  templateUrl: './payroll-charts.component.html',
+  styleUrls: ['./payroll-charts.component.scss'],
+})
+export class PayrollChartsComponent implements OnInit {
+  departmentDistributionData: NameValue[] = [];
+  employeeDistributionData: NameValue[] = [];
+  view: [number, number] = [700, 400];
+
+  // options+
+  gradient = false;
+  showLegend = true;
+  showLabels = false;
+  isDoughnut = false;
+  legendPosition = 'below';
+
+  colorScheme = {
+    domain: ['#5AA454', '#A10A28', '#C7B42C', '#4287f5', '#AAAAAA'],
+  };
+
+  constructor(private dataService: DataService) {
+    this.prepData();
+  }
+
+  ngOnInit(): void {}
+
+  prepData() {
+    this.prepDepartmentDistributionData();
+    this.prepEmployeeDistributionData();
+  }
+
+  prepEmployeeDistributionData() {
+    const payroll = this.dataService.getPayroll();
+    const employees = this.dataService.getEmployees();
+    const map: Map<string, number> = new Map();
+
+    payroll.forEach((p) => {
+      if (p.employeeStatus !== 'ACTIVE') {
+        return;
+      }
+      return map.has(p.employeeName)
```

Comment:
Returning a value in a `.forEach()` doesn't have any meaning. You'd want a `.map()`, `.filter()`, or `.reduce()` if you intend to return a value.